doc/specs/stdlib_math.md

```diff
@@ -0,0 +1,98 @@
+---
+title: math
+---
+
+# The `stdlib_math` module
+
+[TOC]
+
+## Introduction
+
+`stdlib_math` module provides general purpose mathematical functions.
+
+
+## Procedures and Methods provided
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### `clip` function
+
+#### Description
+
+Returns a value which lies in the given interval [`xmin`, `xmax`] (interval is `xmin` and `xmax` inclusive) and is closest to the input value `x`.  
+
+#### Syntax
+
+`res = [[stdlib_math(module):clip(interface)]] (x, xmin, xmax)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument(s)
+
+`x`: scalar of either `integer` or `real` type. This argument is `intent(in)`.  
+`xmin`: scalar of either `integer` or `real` type. This argument is `intent(in)`.  
+`xmax`: scalar of either `integer` or `real` type, which must be greater than or equal to `xmin`. This argument is `intent(in)`.
+
+Note: All arguments must have same `type` and same `kind`.
+
+#### Output value or Result value
+
+The output is a scalar of `type` and `kind` same as to that of the arguments.
+
+#### Examples
+
+##### Example 1:
+
+Here inputs are of type `integer` and kind `int32`
+```fortran
+program demo_clip_integer
+  use stdlib_math
+  use stdlib_kinds
+  implicit none
+  integer(int32) :: x
+  integer(int32) :: xmin
+  integer(int32) :: xmax
+  integer(int32) :: clipped_value
+
+  xmin = -5_int32
+  ! xmin <- -5
+  xmax = 5_int32
+  ! xmax <- 5
+  x = 12_int32
+  ! x <- 12
```

Comment:
I find the comment lines are unneccesary.

Reply:
Should I remove the comment line for `clipped_value` as well?

Reply:
It can stay.